### DISCUSSION TITLE
Why is Theia not considered as an alternative to VS Code?

### DISCUSSION THREAD

**Discussion by ItsCubeTime** [eclipse-theia/theia#9379] [Q&A]
Hi, so Ive been rather bothered by the fact that you cant change editor window layouts in VS Code, came across Theia.

It says on the Theia Blueprint page that its not meant to be used as an IDE as is, are there any missing features in Theia that VS Code has or whats the catch exactly? (To me Theia looks like a just highly modified VS Code branch making frequent merges from VS Code updates)

Im also curious how I could best transfer my settings from vs code to Theia and whether Theia supports .vscode workspaces, tasks.json, settings.json, keybindings.json (etc) out of the box?

**Comment by vince-fugnitto** [comment]
@ItsCubeTime thank you for the discussion! 

>  rather bothered by the fact that you cant change editor window layouts in VS Code, came across Theia

:+1: 

> It says on the Theia Blueprint page that its not meant to be used as an IDE as is, are there any missing features in Theia that VS Code has or whats the catch exactly?

A little bit of information, **Eclipse Theia** is a framework to build cloud and desktop IDE-like applications, and **Theia Blueprint** is one such example. The Theia Blueprint product is fairly new and still in early phase so the disclaimer is currently present. The framework re-uses a lot of components from vscode, but it also allows extension developers the possibility to extend and customize the application to suit the needs and use-cases. The framework of course has not implemented all of the functionality present in vscode as of yet, but we are working towards API-compatibility. We do not fork vscode, but we often copy code when possible.

The vscode similarities are the following:
- support the same monaco editor.
- support LSP, DAP, and TSP (trace server protocol).
- support vscode extensions as plugins (ongoing effort to support new functionality).
- support consuming extensions through the open vscode marketplace alternative (https://open-vsx.org/)
- base implementation of methods, and behavior is consistent with vscode but customizable/extensible through custom `@theia` extensions.
- look and feel of vscode by default.

> I'm also curious how I could best transfer my settings from vs code to Theia and whether Theia supports .vscode workspaces, tasks.json, settings.json, keybindings.json (etc) out of the box?

The framework supports most configurations, but it does not read them from the vscode user home. You can copy all these files under the theia config (`~/.theia`) and it should work.

The `{workspaceName}/.vscode` configurations are supported out of the box.
